src/router.cls.php

```diff
@@ -501,12 +506,30 @@ private function verify_action()
 		// Each action must have a valid nonce unless its from admin ip and is public action
 		// Validate requests nonce (from admin logged in page or cli)
 		if (!$this->verify_nonce($action)) {
-			// check if it is from admin ip
-			if (!$this->is_admin_ip()) {
+			// check if action is from admin ip. skip test for action Core::ACTION_QS_PURGE.
+			if ( $action != Core::ACTION_QS_PURGE && !$this->is_admin_ip()) {
 				Debug2::debug('[Router] LSCWP_CTRL query string - did not match admin IP: ' . $action);
 				return;
 			}
 
+			$save_blog = get_current_blog_id();
+			if ($_REQUEST['switch_blog']) {
```

Comment:
Key may not exist, which will result in warning. Please use `!empty()` to check

Reply:
done

Reply:
This was replaced with new code.